backend/controllers/passController.js

```diff
@@ -0,0 +1,90 @@
+const asyncHandler = require("express-async-handler")
+require("dotenv").config()
+const { connectDB } = require("../config/db")
+const User = require("../model/user.model")
+const Pass = require("../model/pass.model")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+
+//@desc Generate a pass by user
+//@route POST /users/generatePass
+const generatePass = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const { name, email, phone, duration, start } = req.body
+
+    const user = await User.findOne({ _id: req.user._id })
```

Comment:
We had this in req.user. Isn't it? 
`        req.user = await User.findById(decoded.id)` @20 of backend/middleware/authMiddleware.js
Isn't req.user and user defined at line 15 in this file are same?

Reply:
In line 15, we are using findOne, it requires comparison between left and right field, so we've used _id to compare the id of the id in req.user.

In the authMiddleware.js, we are initializing the entire user of that id into req.user

Reply:
Can you please check the values stored in req.user and user(defined @15 line)? and also compare each other.

To compare
1. Stringify both objects  using JSON.stringify(user) and JSON.stringify(req.user)
2. Compare it on any text comparision online website.

Reply:
checked it using 
`    console.log(JSON.stringify(req.user))
    console.log("*******")
    console.log(JSON.stringify(req.user._id))
`
    and got the O/P as
    {"_id":"63a18293500c5ca788a83e29","name":"Random20","email":"random120@gmail.com","password":"$2b$10$tjzdjHitwDfy0cMDVUIItu99RVDgO3r8w.kBHV8xZHuQZgDLhBMTC","phone":"1232132111","roles":["admin"],"createdAt":"2022-12-20T09:38:27.951Z","updatedAt":"2022-12-20T09:38:27.951Z","__v":0}
*******
"63a18293500c5ca788a83e29"

so req.user has got the entire the entire user and req.user._id has got the _id value itself

Reply:
baba... req.user and user(declared at line 13)

Reply:
oh sorry

Reply:
isn't it req.body?

Reply:
that too of a different model, why to compare it? Like it for the person for whom pass is being generated...and the req.user was for the admin

Reply:
yes, both store the same value, i.e. both are identical

Reply:
changed line 15 to 
`    const user = req.user
`

Reply:
Is this really required? Can we use req.user directly as we have done it at line 25.

Reply:
yes, we can, line 15 removed